pkg/api/addons.go

```diff
@@ -51,7 +59,7 @@ func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
 
 	defaultACIConnectorAddonsConfig := KubernetesAddon{
 		Name:    ACIConnectorAddonName,
-		Enabled: to.BoolPtr(DefaultACIConnectorAddonEnabled),
+		Enabled: to.BoolPtr(DefaultACIConnectorAddonEnabled && !cs.Properties.IsAzureStackCloud()),
```

Comment:
Carried this additional default gate over from the "last mile" "concatenate addons string" business logic area.